firebase-firestore/src/test/java/com/google/firebase/firestore/core/QueryTest.java

```diff
@@ -553,4 +559,62 @@ public void testMatchesAllDocuments() {
     query = baseQuery.endAt(new Bound(Collections.emptyList(), true));
     assertFalse(query.matchesAllDocuments());
   }
+
+  @Test
+  public void testCanonicalIdsAreStable() {
+    // This test aims to ensure that we do not break canonical IDs, as they are used as keys in
+    // the TargetCache.
+
+    Query baseQuery = Query.atPath(ResourcePath.fromString("collection"));
+
+    Map<Query, String> queryToCanoncialIdMap = new LinkedHashMap<>();
+    queryToCanoncialIdMap.put(baseQuery, "collection|f:|ob:__name__asc|lt:LIMIT_TO_FIRST");
```

Comment:
Consider adding an assertion helper instead. Something like

```java
private void assertCanonicalId(String canonicalId, Query query) {
  assertEquals(canonicalId, query.getCanonicalId());
}
```
These end up being about the same size at the call site but are easier to step through/debug/see exactly what failed just from exception line numbers.

Reply:
Done